### PR TITLE
chore: remove dead code, add geometry/fchk tests, CI coverage

### PR DESCRIPTION
**Pull request by ericchansen**
## Summary

Remove dead code, improve documentation, add missing tests, and wire up coverage reporting in CI.

### Changes

**Dead code removal**
- Remove `_GEOM_NONCONV_PENALTY` constant from `jaxloss.py` — was hardcoded to `0.0`, so the penalty branch (`0.0 * n_geom_refs`) always contributed zero. Also removes the now-unused `n_geom_refs` computation from `_build()`.

**Documentation**
- Add docstring for `_MM3_TYPE_MAP` in `diagnostics/systems.py` explaining the element + bond-count → MM3 type mapping convention, with examples and literature reference.

**New tests**
- `test_geometry.py` — 14 tests covering `bond_length`, `bond_angle`, `dihedral_angle` (edge cases: collinear, sign convention, symmetry, numpy arrays).
- `test_io_fchk.py` — 10 tests covering `parse_fchk` with ethane GS/TS fixtures (symbols, coords shape, Angstrom units, Hessian symmetry, TS negative eigenvalue, error paths).

**CI coverage**
- Add `pytest-cov` to dev dependencies.
- Add `--cov=q2mm --cov-report=term-missing --cov-report=xml` to the `test-core` CI job.
- Upload `coverage.xml` as an artifact from the Python 3.12 matrix entry.

### No behavioral changes
All modifications are additive (tests, docs, CI) or remove provably dead code.

### CODE REVIEW

File: test/test_io_fchk.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+@pytest.fixture()
+def ethane_gs() -> FchkResult:
+    """Parse the ethane ground-state .fchk fixture."""
+    return parse_fchk(_ETHANE_GS)
+
+
+@pytest.fixture()
+def ethane_ts() -> FchkResult:
+    """Parse the ethane transition-state .fchk fixture."""
+    return parse_fchk(_ETHANE_TS)
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply test/test_io_fchk.py]: Fixed in 47445b1. Replaced the local `Path(__file__)...` derivation with `from test._shared import GS_FCHK, TS_FCHK` — now uses the central fixture path constants alongside the existing fchk tests in the codebase.

File: q2mm/optimizers/jaxloss.py [hunk 2]
```diff
@@ -449,12 +437,7 @@ def _geometry_loss_from_entry(entry: dict, params: np.ndarray) -> np.ndarray:
             mol_spec = entry["mol_spec"]
             energy_fn = handle._energy_fn
 
-            relaxed, geom_converged = _relax_coords(energy_fn, params, coords)
-            total = total + jnp.where(
-                geom_converged,
-                0.0,
-                _GEOM_NONCONV_PENALTY * entry["n_geom_refs"],
-            )
+            relaxed, _geom_converged = _relax_coords(energy_fn, params, coords)
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply q2mm/optimizers/jaxloss.py:445]: Good catch — fixed in 47445b1. Dropped the unused `converged` return from `_relax_coords` entirely and rewrote its docstring to explain the history (penalty zeroed in 8c56fe9 then removed in this PR). Call site updated to single-value return. Callers needing convergence diagnostics can still inspect `solver.state.error` directly — pointed to q2mm#284 for the broader noise discussion.
